### DISCUSSION TITLE
Best practice for AuthN and AuthZ

### DISCUSSION THREAD

**Discussion by margaretjoanmiller** [async-graphql/async-graphql#1615] [Q&A]
I was wondering how folks are handeling authentication and authorization. Guards? Scheme directives?

I'm using a jwt if that matters.

**Comment by emanuel-gjini** [comment]
I have a tiny project that I work on it from time to time whenever I get a chance and I'm using poem and its session module (poem::session) to handle the authentication based on a signed & encrypted cookie that holds various info like user_id and other metadata with a 24h expiration time.

Then the session obj along with a custom struct that holds various other objs like DB connection pool, env vars etc are passed as context data to the schema. On the schema side I use guards to protect the queries/fields I want and leave some others public like the "signup" mutation or the "login" query.

Hope that helps.
